hbase-it/pom.xml

```diff
@@ -408,6 +408,10 @@
           <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-minikdc</artifactId>
         </dependency>
+        <dependency>
```

Comment:
Usually we do not want to include hbase-testing-util in other modules? What is the actual problem here? Dependency conflicts or we miss some dependencies here?

Reply:
```com.sun.jersey:jersey-core``` is missing as it is being excluded in pom.xml. Also ```hbase-testing-util ``` is already a dependency to this project and it excludes ```com.sun.jersey:jersey-core``` which is what this PR is trying to override by defining the ```hbase-testing-util ``` w/o exclusion on ```com.sun.jersey:jersey-core```. See https://github.com/apache/hbase/blob/7016c640960dd875f2daf78601dc68bfe14a1219/hbase-it/pom.xml#L96

Reply:
I think the exclusion is done via HBASE-22052? Seems it is required by MiniMRYarnCluster? How do we handle this problem in other modules, such as hbase-mapreduce?

Reply:
I checked there earlier. We are not excluding this dependency which is coming via hadoop-minicluster and hence tests run fine there

Reply:
Then which commit breaks these ITs?

Reply:
Let me dig a little deeper and come back.

Reply:
> I think the exclusion is done via [HBASE-22052](https://issues.apache.org/jira/browse/HBASE-22052)?

The culprit is indeed [HBASE-22052](https://issues.apache.org/jira/browse/HBASE-22052). Did following to verify:
1) Look at commit history for hbase-it. See https://github.com/apache/hbase/commits/branch-2/hbase-it?after=9878b14906320ba6f5ab932cdc51e3e454a82cbd+69

2) Checkout commit for HBASE-22052, build code and run test: FAIL
```
git checkout 939a29b41e0df228891e04b38c109042b8531b59 
mvn clean install -DskipTests 
mvn --no-snapshot-updates -Dtest=IntegrationTestImportTsv -pl hbase-it package
```
3) Checkout  commit just before HBASE-22052 in hbase-it, build code and run test: PASS
```
git checkout 939a29b41e0df228891e04b38c109042b8531b59 
mvn clean install -DskipTests 
mvn --no-snapshot-updates -Dtest=IntegrationTestImportTsv -pl hbase-it package
```

Reply:
I think HBASE-22052 is for fixing running IT against a real cluster, where we do not need to start a mini cluster in process? But anyway, I think if we can run the IT with a mini cluster, it does not make sense that we can not run it with a real cluster...

SO let's just remove the exclusion to see if things go right?

Reply:
Then better move the hbase-testing-util dependency to the hadoop2 and hadoop3 profiles? For hadoop3, we exclude the transitive dependency while in hadoop2 we do not?

Reply:
Makes sense. Let me do that and update the PR.